Plutarch/Builtin.hs

```diff
@@ -1,4 +1,9 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE QuantifiedConstraints #-}
+{-# LANGUAGE RoleAnnotations #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
 
 -- This should have been called Plutarch.Data...
```

Comment:
?

Reply:
It probably makes sense to rename this. Where should we put `PBuiltinList`?

Reply:
`Plutarch.Builtin` seems fine and correct to me. The comment seems outdated.